src/main/python/systemml/converters.py

```diff
@@ -212,10 +215,86 @@ def convertToNumPyArr(sc, mb):
     else:
         raise TypeError('sc needs to be of type SparkContext') # TODO: We can generalize this by creating py4j gateway ourselves
 
+
+#   Load an image converting from grayscale or alpha as needed.
+def load_image(filename, color=True):
+    """
+    Input Parameters
+    ----------
+    filename : string
+    color : boolean
+        flag for color format. True (default) loads as RGB while False
+        loads as intensity (if image is already grayscale).
+
+    Returns
+    -------
+    image : an image with type np.float32 in range [0, 1]
+        of size (H x W x 3) in RGB or
+        of size (H x W x 1) in grayscale.
+    """
+    import cv2
+    image = skimage.img_as_float(cv2.imread(filename))
+    # cv2 reads in BGR format, so converting back to RGB format.
+    image = image[:,:,::-1]
+    if image.ndim == 2:
+        image = image[:, :, np.newaxis]
+        if color:
+            image = np.tile(image, (1, 1, 3))
+    elif image.shape[2] == 4:
+        image = image[:, :, :3]
+    return image
+
+
+# Resize an image array with interpolation.
+def resize_image(image, new_dims, interp_order=1):
+    """
+    Input Parameters
+    ----------
+    image : (H x W x K) ndarray
+    new_dims : (height, width) tuple of new dimensions.
+    interp_order : interpolation order, default is linear.
+
+    Returns
+    -------
+    image : resized ndarray with shape (new_dims[0], new_dims[1], K)
+    """
+    if image.shape[-1] == 1 or image.shape[-1] == 3:
+        im_min, im_max = image.min(), image.max()
+        if im_max > im_min:
+            # skimage is fast but only understands {1,3} channel images
+            # in [0, 1].
+            im_std = (image - im_min) / (im_max - im_min)
+            resized_std = resize(im_std, new_dims, order=interp_order)
+            resized_im = resized_std * (im_max - im_min) + im_min
+        else:
+            # the image is a constant -- avoid divide by 0
+            ret = np.empty((new_dims[0], new_dims[1], image.shape[-1]),
+                           dtype=np.float32)
+            ret.fill(im_min)
+            return ret
+    else:
+        # ndimage interpolates anything but more slowly.
+        scale = tuple(np.array(new_dims, dtype=float) / np.array(image.shape[:2]))
+        resized_im = zoom(image, scale + (1,), order=interp_order)
+    return resized_im.astype(np.float32)
+
+
 # Example usage: convertImageToNumPyArr(im, img_shape=(3, 224, 224), add_rotated_images=True, add_mirrored_images=True)
 # The above call returns a numpy array of shape (6, 50176) in NCHW format
-def convertImageToNumPyArr(im, img_shape=None, add_rotated_images=False, add_mirrored_images=False):
-    from PIL import Image
+def convertImageToNumPyArr(image_file_name, img_shape=None, add_rotated_images=False, add_mirrored_images=False,
+    color=True, transpose=(2,0,1), color_mode = 'BGR', raw_scale=255, mean=[103.939, 116.779, 123.68]):
```

Comment:
yes, that was for short term until I address other models, but anyway changed it.